tests/internal/aws_s3_local_buffer.c

```diff
@@ -0,0 +1,154 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+#include <fluent-bit/flb_sds.h>
+#include <fluent-bit/flb_s3_local_buffer.h>
+#include <fluent-bit/flb_output_plugin.h>
+#include <fluent-bit/flb_mem.h>
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_aws_util.h>
+
+#include "flb_tests_internal.h"
+
+#define DIRECTORY "/fluent-bit/buffer/s3"
+#define PLUGIN_NAME "flb_s3_plugin"
+#define TEST_DATA "I love Fluent Bit"
+#define KEY_1 "key1"
+#define KEY_2 "key2"
+#define KEY_3 "key3"
+
+static void test_chunk_compare(struct flb_local_buffer *store, 
+                               struct flb_local_chunk *chunk,
+                               char *tag, char *data)
+{
+    char *buffered_data = NULL;
+    size_t buffer_size;
+    struct mk_list *tmp;
+    struct mk_list *head;
+    int ret;
+    /* Ensure data retreived is same as that which was stored. */
+    mk_list_foreach_safe(head, tmp, &store->chunks) {
```

Comment:
Actually... why iterate over the list of chunks if you already have the chunk?

Reply:
Thanks for pointing this out! I do have the chunk I can just compare it here. There is no need to iterate over all the chunks.